ocean_provider/utils/currency.py

```diff
@@ -31,29 +29,58 @@
 """The maximum possible token amount on Ethereum-compatible blockchains, denoted in ether"""
 MAX_ETHER = Decimal(MAX_WEI).scaleb(-18, context=ETHEREUM_DECIMAL_CONTEXT)
 
+UNITS = [
```

Comment:
I'm not a fan of adding these here. Is there some library constant we ca use instead?

Reply:
eth-utils currency.py has a `units` dictionary that maps names to their corresponding `Decimal` value. https://github.com/ethereum/eth-utils/blob/d8b5aa197c3bf6d9d23b677556373f44a6aa3320/eth_utils/units.py#L6-L30

Using `units` to calculate the exponent requires logarithms (which admittedly, I'm rusty at) so at the time, just copying the `UNITS` list from ethers.js was faster. 

But now that I'm revisiting, I took the time to figure out the logarithm logic which will allow me to remove the `UNITS` list.

```python
    num_decimals = (
        int(units[unit_name].log10()) if isinstance(unit_name, str) else unit_name
    )
```

NOTE:  I chose not to use `dict.get()` because I think this method should throw an exception if the user passes in an unsupported unit name.

The corresponding ocean.py PR has already been merged: https://github.com/oceanprotocol/ocean.py/pull/784 so I'll create a new PR in ocean.py with this fix.